### PR TITLE
uplink: support server-side move operation

### PR DESCRIPTION
**Pull request by elek**
This patch expose the new move command from uplink-c.

### CODE REVIEW

**Review by Erikvv — APPROVED** [review]
Test isn't running in CI but looks good otherwise.

**Comment by elek** [comment]
> Test isn't running in CI but looks good otherwise.

They seem to be executed: https://github.com/storj/uplink-java/actions/runs/2281601070

And checked the junit file and the new tests are executed:

```
  60   │   <testcase name="testMoveFailure" classname="io.storj.UplinkTest" time="0.052"/>
  61   │   <testcase name="testObjectsListing" classname="io.storj.UplinkTest" time="1.922"/>
  62   │   <testcase name="testMove" classname="io.storj.UplinkTest" time="0.93"/>
```

File: src/main/java/io/storj/Project.java [hunk 1]
```diff
@@ -126,6 +126,20 @@ public ObjectInfo statObject(String bucket, String key) throws StorjException {
...
     }
 
+    public void moveObject(String oldBucket, String oldKey, String newBucket, String newKey) throws StorjException {
+        JNAUplink.Error.ByReference error = JNAUplink.INSTANCE.uplink_move_object(this.project, oldBucket, oldKey, newBucket, newKey);
+        try {
+            ExceptionUtil.handleError(error);
+        } finally {
+            if (error != null) {
+                // autoread is required here to avoid re-reading the fields which are already free-d by the uplink-c.
+                error.setAutoRead(false);
```
Comment (mniewrzal) [inline src/main/java/io/storj/Project.java:136]: should we also apply this in other places like this?

**Review by mniewrzal — APPROVED** [review]

**Comment by Erikvv** [comment]
> > Test isn't running in CI but looks good otherwise.
> 
> They seem to be executed: https://github.com/storj/uplink-java/actions/runs/2281601070
> 
> And checked the junit file and the new tests are executed:
> 
> ```
>   60   │   <testcase name="testMoveFailure" classname="io.storj.UplinkTest" time="0.052"/>
>   61   │   <testcase name="testObjectsListing" classname="io.storj.UplinkTest" time="1.922"/>
>   62   │   <testcase name="testMove" classname="io.storj.UplinkTest" time="0.93"/>
> ```

I was confused because there isn't a "test" step in https://github.com/storj/uplink-java/runs/6322434041?check_suite_focus=true and none of the output or commands looks like any tests are run.

**Comment by elek** [comment]
@Erikvv Good point. 

It doesn't worth to separate build/test IMHO (test includes build), but it's a good idea to rename the stage to `test` to make it clear what's going on...

**Pull request merged**